src/factory/src/api/inspect_message.rs

```diff
@@ -0,0 +1,19 @@
+use crate::state::State;
+use ic_cdk_macros::inspect_message;
+use ic_storage::IcStorage;
+
+#[cfg(not(feature = "no_api"))]
+#[inspect_message]
+fn inspect_message() {
+    let state = State::get();
+    let state = state.borrow();
+
+    if ic_cdk::api::call::method_name() == "set_token_bytecode" {
```

Comment:
This method needs to check that the caller is the owner, not everyone should be able to set the token bytecode.

Reply:
It should, but we do not even store owner in state in first place

Reply:
didn't want to mess with this logic since the factory is used in another crates

Reply:
ah, no, it's just named differently